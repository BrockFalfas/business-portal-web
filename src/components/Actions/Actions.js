import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './Actions.scss';

const Left = ({ children }) => <div className="Actions-left">{children}</div>;
const Right = ({ children }) => <div className="Actions-right">{children}</div>;

export class Actions extends React.Component {
  static Left = Left;
  static Right = Right;

  static propTypes = {
    position: PropTypes.oneOf(['top', 'bottom']).isRequired,
  };

  render() {
    const { position, children } = this.props;
    return (
      <div
        className={classnames('Actions', {
          [`Actions--${position}`]: true,
        })}>
        {children}
      </div>
    );
  }
}

export default Actions;
