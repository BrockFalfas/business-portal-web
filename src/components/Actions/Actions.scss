@import "../../styles/var";

.Actions {
  display: flex;
  align-items: center;
  justify-content: space-between;

  &--top {
    margin-bottom: 1rem;
  }

  &--bottom {
    margin-top: 1rem;
  }
}

.Actions-left {
  flex-grow: 1;
  text-align: left;

  > * + * {
    margin-left: 1rem;
  }
}
.Actions-right {
  flex-grow: 1;
  text-align: right;

  > * + * {
    margin-right: 1rem;
  }
}

@media (max-width: $screen-md-max) {
  .Actions {
    flex-wrap: wrap;
  }
  .Actions-left,
  .Actions-right {
    margin-bottom: 0.5rem;
  }
}

@media (max-width: $screen-sm-max) {
  .Actions-left,
  .Actions-right {
    text-align: left;
    flex-basis: 100%;
  }
}
