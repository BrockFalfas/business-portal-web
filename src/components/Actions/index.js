import Actions from './index';

export default Actions;
