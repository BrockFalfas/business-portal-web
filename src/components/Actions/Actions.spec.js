import React from 'react';
import { shallow } from 'enzyme';

import Actions from './index';

describe('component: <Actions/>', () => {
  describe('Actions', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<Actions position="top" />);
      expect(wrapper).toBeTruthy();
    });

    it('should render children properly', () => {
      const fakeChild = <div>fake child</div>;
      const wrapper = shallow(<Actions position="top">{fakeChild}</Actions>);
      expect(wrapper).toMatchSnapshot();
    });

    it('should use proper positioning', () => {
      let wrapper = shallow(<Actions position="top" />);
      expect(wrapper.prop('className')).toContain('Actions--top');

      wrapper = shallow(<Actions position="bottom" />);
      expect(wrapper.prop('className')).toContain('Actions--bottom');
    });
  });

  describe('Actions.Left', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<Actions.Left />);
      expect(wrapper).toBeTruthy();
    });

    it('should render children properly', () => {
      const fakeChild = <div>fake child</div>;
      const wrapper = shallow(<Actions.Left>{fakeChild}</Actions.Left>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Actions.Right', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<Actions.Right />);
      expect(wrapper).toBeTruthy();
    });

    it('should render children properly', () => {
      const fakeChild = <div>fake child</div>;
      const wrapper = shallow(<Actions.Right>{fakeChild}</Actions.Right>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
