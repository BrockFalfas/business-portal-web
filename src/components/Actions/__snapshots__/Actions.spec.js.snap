// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Actions/> Actions should render children properly 1`] = `
<div
  className="Actions Actions--top"
>
  <div>
    fake child
  </div>
</div>
`;

exports[`component: <Actions/> Actions.Left should render children properly 1`] = `
<div
  className="Actions-left"
>
  <div>
    fake child
  </div>
</div>
`;

exports[`component: <Actions/> Actions.Right should render children properly 1`] = `
<div
  className="Actions-right"
>
  <div>
    fake child
  </div>
</div>
`;
